### PR TITLE
make filter docs use fluent syntax by default

### PR DESCRIPTION
**Pull request by tabacitu**
I've spent quite a bit of time rewriting this page, so that:
- it uses the fluent syntax
- it is simpler to read and understand

I need help TESTING the docs here. I need someone to use the code I provided in an app, and make sure it's working they're all working expected.

@karandatwani92 can you help with that please?

-- 

Addresses a small part of https://github.com/Laravel-Backpack/docs/issues/450

### CODE REVIEW

File: 6.x/crud-filters.md [hunk 1]
```diff
@@ -5,63 +5,176 @@
...
 
-// aditionally you should be able to change Filter attributes by appending `->methods` in your filter definition. Eg - this would set the filter attribute `type` to be equal to `number`:
-$this->crud->addFilter($options, $values, $filter_logic)->type('number');
+```php
+CRUD::filter('name')
+    ->type('text')
+    ->label('The name')
+    ->whenActive(function($value) {
+        CRUD::addClause('where', 'name', 'LIKE', '%'.$value.'%');
+    })->else(function($value) {
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:51]: @tabacitu else closure causing an error `Too few arguments to function {closure}`

File: 6.x/crud-filters.md [hunk 2]
```diff
@@ -5,63 +5,176 @@
...
+- ```->remove()``` - By chaining **remove()** on a filter you remove it from the current operation;
+
+```php
+CRUD::filter('name')->remove();
+```
+
+- ```->forget('attribute_name')``` - By chaining **forget('attribute_name')** to a filter you remove that attribute from the filter;
+
+```php
+CRUD::filter('name')->forget('suffix');
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:131]: @tabacitu 
- `suffix` attribute is not used in any filter. Applicable example make more sense.
- forget not working for the `before` attribute. & When I tried to the `forget('label')` This made the label empty, where it is supposed to pick from filter `name`
- No idea how it works with `ifActive` OR `whenActive` (two-word-attribute)
- `CRUD::filter('name)->label('Full Name');` is missing in Other Chained Methods section

File: 6.x/crud-filters.md [hunk 3]
```diff
@@ -5,63 +5,176 @@
...
+
+Backpack filters do not contain any default filtering _logic_, because it cannot infer that. If you use a filter and don't specify a filter logic, no filtering of entries will actually happen. You have to define that, inside a _closure_.
+
+Filter logic closures are just an anonymous functions that gets executed when the filter is active. You can use any Laravel or Backpack functionality you want inside them. For example, a valid closure would be:
+
+```php
+(function($value) {
+  CRUD::addClause('where', 'draft', 1);
+})
+```
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:169]:
```php
CRUD::filter('draft')
 ->whenActive(function($value) {
  CRUD::addClause('where', 'draft', 1);
 })
```
I would prefer to have a complete code block. Easier to copy and paste for the user.

File: 6.x/crud-filters.md [hunk 4]
```diff
@@ -5,63 +5,176 @@
...
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	})->apply();
+
+// whenActive, left to be applied by the operation itself
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:88]: Replace with `CRUD::addClause`

File: 6.x/crud-filters.md [hunk 5]
```diff
@@ -5,63 +5,176 @@
...
+- ```->whenInactive(function($value) {})``` - By chaining **whenInactive()** on a filter you define what should be done when that filter is NOT active; it accepts a closure that is called when the filter is NOT active - either immediately by further chaining ```apply()``` on the filter, or automatically after all filters are defined, by the List operation itself; you can also use ```->else()```, ```->fallbackLogic()```, ```->whenNotActive()```, ```->ifInactive()``` and ```->ifNotActive()``` which are its aliases:
+
+```php
+// fallbackLogic method, applied immediately
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
+	})->whenInactive(function ($value) {
+	    $this->crud->addClause('where', 'active', '0');
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:101]: Replace with `CRUD::addClause` x2

File: 6.x/crud-filters.md [hunk 6]
```diff
@@ -74,16 +187,11 @@ Only shows a label and can be toggled on/off. Useful for things like active/inac
...
-  'label' => 'Active'
-],
-false,
-function() { // if the filter is active
-  // $this->crud->addClause('active'); // apply the "active" eloquent scope
-} );
+CRUD::filter('active')
+    ->type('simple')
+    ->whenActive(function() {
+      // $this->crud->addClause('active'); // apply the "active" eloquent scope
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:193]: Replace with `CRUD::addClause`

File: 6.x/crud-filters.md [hunk 7]
```diff
@@ -96,16 +204,11 @@ Shows a text input. Most useful for letting the user filter through information
...
-  'label' => 'Description'
-],
-false,
-function($value) { // if the filter is active
-  // $this->crud->addClause('where', 'description', 'LIKE', "%$value%");
-});
+CRUD::filter('description')
+    ->type('text')
+    ->whenActive(function($value) {
+      // $this->crud->addClause('where', 'description', 'LIKE', "%$value%");
```
Comment (karandatwani92) [inline 6.x/crud-filters.md:210]: Replace with `CRUD::addClause`

File: 6.x/crud-filters.md [hunk 8]
```diff
@@ -5,63 +5,176 @@
...
+```
+
+- ```->whenActive(function($value) {})``` - By chaining **whenActive()** on a filter you define what should be done when that filter is active; it accepts a closure that is called when the filter is active - either immediately by further chaining ```apply()``` on the filter, or automatically after all filters are defined, by the List operation itself; you can also use ```->logic()``` or ```->ifActive()``` which are its aliases:
+
+```php
+// whenActive method, applied immediately
+CRUD::filter('active')
+	->type('simple')
+	->whenActive(function ($value) {
+	    $this->crud->addClause('where', 'active', '1');
```
Comment (pxpm) [inline 6.x/crud-filters.md:81]:
```suggestion
	    CRUD::addClause('where', 'active', '1');
```

File: 6.x/crud-filters.md (hunk 2 above)
Comment (tabacitu) [reply 6.x/crud-filters.md:131]:
```suggestion
CRUD::filter('price')->prefix('$'); // will have "$" as prefix 
CRUD::filter('price')->forget('prefix'); // will no longer have "$" as prefix
```

Comment (tabacitu) [reply 6.x/crud-filters.md:131]: > forget not working for the before attribute. & When I tried to the forget('label') This made the label empty, where it is supposed to pick from filter name
> No idea how it works with ifActive OR whenActive (two-word-attribute)

That makes sense. Forget will only forget attributes, it won't work on methods or aliases. 
- `before()` isn't an attribute, it's a method;
- `whenActive()` isn't an attribute, it's a method, an alias to `logic()`
- `whenInactive()` isn't an attribute, it's a method, an alias to `fallbackLogic()`

So:
- before cannot be forgotten
- logic/fallbacklogic can be forgotten with their attributes `logic` / `fallbackLogic`

You think we should document this here, as a note?

Comment (tabacitu) [reply 6.x/crud-filters.md:131]: I updated my suggestion above to cover this, let me know what you think.

File: 6.x/crud-filters.md (hunk 3 above)
Comment (tabacitu) [reply 6.x/crud-filters.md:169]: Sounds good. Heads-up, you can use the Github suggest button here, to make a suggestion. Then it'll be easy for the reviewer to add your code to the codebase:

CleanShot 2023-08-11 at 15 19 14@2x

File: 6.x/crud-filters.md [hunk 9]
```diff
@@ -5,63 +5,176 @@
...
+#### Filter logic closures
+
+Backpack filters do not contain any default filtering _logic_, because it cannot infer that. If you use a filter and don't specify a filter logic, no filtering of entries will actually happen. You have to define that, inside a _closure_.
+
+Filter logic closures are just an anonymous functions that gets executed when the filter is active. You can use any Laravel or Backpack functionality you want inside them. For example, a valid closure would be:
+
+```php
+(function($value) {
+  CRUD::addClause('where', 'draft', 1);
+})
```
Comment (tabacitu) [inline 6.x/crud-filters.md:168]:
```suggestion
CRUD::filter('draft') ->whenActive(function($value) {
    CRUD::addClause('where', 'draft', 1);
});
```

File: 6.x/crud-filters.md (hunk 3 above)
Comment (pxpm) [reply 6.x/crud-filters.md:169]: > Sounds good. Heads-up, you can use the Github suggest button here, to make a suggestion. Then it'll be easy for the reviewer to add your code to the codebase:
> 
> CleanShot 2023-08-11 at 15 19 14@2x

Too late:
image

Comment (tabacitu) [reply 6.x/crud-filters.md:178]: Haha perfect. Closing then.

File: 6.x/crud-filters.md (hunk 1 above)
Comment (tabacitu) [reply 6.x/crud-filters.md:51]: Oh that's odd... what's the solution here @karandatwani92

File: 6.x/crud-filters.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (karandatwani92) [reply 6.x/crud-filters.md]: > Search `$this->crud->addClause` & **Replace** with `CRUD::addClause` **in whole document**

It's still there 38 times. NOT RESOLVED

File: 6.x/crud-filters.md (hunk 1 above)
Comment (karandatwani92) [reply 6.x/crud-filters.md:51]: This works perfectly without $value `->else(function() { }` and it's right.

My test says:
->withActive() //$value is passed
->else() //$value not passed

->whenInActive() //$value not passed
->else() //$value is `null` passed.

File: 6.x/crud-filters.md (hunk 2 above)
Comment (karandatwani92) [reply 6.x/crud-filters.md]: @tabacitu Everything is perfect here but no filter has a prefix attribute

File: 6.x/crud-filters.md (hunk 1 above)
Comment (tabacitu) [reply 6.x/crud-filters.md:51]: Ok thank you

```suggestion
    })->else(function() {
```

**Pull request merged**